### ISSUE TITLE
Set-MailboxMessageConfiguration has no effect

### ISSUE THREAD

**Issue by leewisestamp** [MicrosoftDocs/office-docs-powershell#8537]
We have been successfully updating OWA signatures using `Set-MailboxMessageConfiguration` for the past 6+ months. Suddenly, and with no change on our part, `Set-MailboxMessageConfiguration` has started to have no effect. Our business heavily depends on this cmdlet and we would really appreciate your feedback/guidance. Someone else posted the same issue on SO here: https://stackoverflow.com/q/69764573/152630.

Example (that previously worked):
```
Set-MailboxMessageConfiguration -Identity "joe@domain.com" -SignatureHTML "My new sig" -AutoAddSignature $True -AutoAddSignatureOnReply $True
```

Interestingly, after manually running `Set-MailboxMessageConfiguration` in a local PowerShell, we see the correct result in `Get-MailboxMessageConfiguration` (https://www.screencast.com/t/X9i4BY1jSb) but NOT in the OWA UI. Please advise.

**Comment by leewisestamp** [comment]
Any guidance on the matter would be greatly appreciated

**Comment by chrisda** [comment]
@leewisestamp, the best way for you to get an answer is to open a support ticket. The fact that this feature worked fine and then suddenly stopped working indicates some kind of problem that's beyond documentation.

Report back here with your findings.

**Comment by leewisestamp** [comment]
@chrisda Obviously, we did that in parallel to opening this documentation ticket. So far Support has no answers other than a shaky workaround that will effectively ruin our product.

And I disagree that this is beyond documentation because if the cmdlet behavior has changed (e.g. to support multiple signatures in OWA) without that being documented (e.g. by adding a new parameter to the cmdlet to specify which signature should be updated), then that is a gap in documentation.

**Comment by chrisda** [comment]
@leewisestamp, the issue is determining whether this is a deliberate feature change or some transitory problem that's going to get fixed. The answer to those questions determines whether we update documentation or not.

So is your ticket still open? Are they looking into it and gave you the lame workaround as a stop-gap until they can find out what's going on? Or did they say "that's the way it is" and close your ticket? And are you saying that there are new undocumented parameters/behavior that need to be documented?

**Comment by leewisestamp** [comment]
@chrisda I don't work for MSFT so your team is far better qualified to determine if this deliberate or a bug.

Our ticket is still open with effectively no progress for 2 days other than the lame workaround and "we're looking into it".

As regards undocumented OWA behavior, multiple test accounts of ours inexplicably received the following "multiple signatures" functionality which we can't find documented anywhere.
![image](https://user-images.githubusercontent.com/11390801/140382597-a037fa5a-ac89-42a2-8ab4-3ef6542eb225.png)

My comment about a potentialy discrepancy between `Set-MailboxMessageConfiguration` and OWA is pure speculation. Again, I don't work for MSFT and I have no official or unofficial documentation supporting my speculation other than the problem we're seeing (and which a couple of others have now reported on StackOverflow).

> Also, where should a bug in the cmdlet be reported if not here?

**Comment by wojcib** [comment]
In my organization we also depend heavily Set-MailboxMessageConfiguration cmdlet.  @leewisestamp probably found something here. In my organization there are users that have the functionality with multiple signature and these are the ones where cmdlet doesn't work anymore. Mailboxes of users that have 'old' functionality with only one single signature, are working properly (signatures and settings are updating with cmdlet).

**Comment by chrisda** [comment]
@wojcib, understood. Attempting to find answers. If you haven't already, I would also suggest opening a support ticket.

**Comment by David-Forlen-Themis** [comment]
It's not limited to this project : we're also impacted when using the Dictionary of `UserConfiguration` in C# .NET (following instructions from https://social.msdn.microsoft.com/Forums/en-US/0b654e7f-2206-44bb-a595-7eed4c299c5b/how-to-create-an-owa-email-signature-by-using-ews?forum=exchangesvrdevelopment).

We're in the same position as you : the code does update the configuration for the user on OWA, but this isn't reflected in the new UI.

Maybe Microsoft decided to store its signatures elsewhere ?
@leewisestamp,  I'm interessed if you found anything to address this issue.

**Comment by leewisestamp** [comment]
Any insights? Thanks in advance. To be clear, this is very urgent for us.

**Comment by koraniar** [comment]
Any update of this? my project in C# was also affected in the same way as @David-Forlen-Themis

**Comment by chrisda** [comment]
No updates. I strongly encourage any of you who haven't already to open a support ticket about this.

**Comment by andreaci** [comment]
I noticed the same problem today; opened a new ticket

**Comment by leewisestamp** [comment]
> I noticed the same problem today; opened a new ticket

Any progress with your ticket? Our Premium support ticket has been opened for nearly two weeks with the only reply being "we're still looking into it".

**Comment by andreaci** [comment]
> > I noticed the same problem today; opened a new ticket
> 
> Any progress with your ticket? Our Premium support ticket has been opened for nearly two weeks with the only reply being "we're still looking into it".

it's the first ticket in history where I've not been called back in a few minutes.
Stilll no answers

**Comment by dcastelli77** [comment]
Same issue here. First support ticket opened 1 month ago. Support had not workaround neither resolution. Some users have multiple signature feature enabled and for these users cmdlet set-mailboxmessageconfiguration does not work. Other ticket opened last week. No reply from Microsoft.

**Comment by dcastelli77** [comment]
> > I noticed the same problem today; opened a new ticket
> 
> Any progress with your ticket? Our Premium support ticket has been opened for nearly two weeks with the only reply being "we're still looking into it".

No words!

**Comment by dcastelli77** [comment]
> No updates. I strongly encourage any of you who haven't already to open a support ticket about this.

thank you for suggestion but the opened tickets have no reply from Microsoft.

**Comment by chrisda** [comment]
Please don't let the lack of response scare you from opening a support ticket. Volume is important here. The more people that open tickets about the issue, the higher the escalation likely will be.

**Comment by dcastelli77** [comment]
> Please don't let the lack of response scare you from opening a support ticket. Volume is important here. The more people that open tickets about the issue, the higher the escalation likely will be.

I have opened the ticket but Microsoft does not reply. (for the same issue I have opened 3 tickets).
thank you!

**Comment by ricardopolo** [comment]
Microsoft is closing the tickets saying they don't support scripts

**Comment by chrisda** [comment]
I would rephrase the issue:

> Any mailbox that allows the configuration of multiple signatures in Outlook on the Web no longer recognizes or uses the SignatureHTML value that's configured on the mailbox using the Set-MailboxMessageConfiguration cmdlet like it used to (for years). So where are the multiple Outlook on the Web signatures now stored, and how can we manage them using PowerShell cmdlets?

That's the fundamental issue.

**Comment by beail** [comment]
If the user Deletes there current signature from Outlook Online and ensures both For New Messages, For Replies/Forwards = (No signature) this then oddly falls back to whatever is defined in the SignatureHTML property

**Comment by wojcib** [comment]
> If the user Deletes there current signature from Outlook Online and ensures both For New Messages, For Replies/Forwards = (No signature) this then oddly falls back to whatever is defined in the SignatureHTML property

I checked and your observations are correct. But it only works when user clicks 'Delete' (right above signature) from user's OWA interface. 
I tried to script that somehow but had no luck.

**Comment by beail** [comment]
> > If the user Deletes there current signature from Outlook Online and ensures both For New Messages, For Replies/Forwards = (No signature) this then oddly falls back to whatever is defined in the SignatureHTML property
> 
> I checked and your observations are correct. But it only works when user clicks 'Delete' (right above signature) from user's OWA interface. I tried to script that somehow but had no luck.

There's a user element at play here; we _maybe_ able to determine the graph call used via that 'Delete' action using the Edge development tools but I'm not sure it'll be something we could interface with without a user token (auth) but this is speculative as I've not had an opportunity to investigate this potential approach yet - maybe worth a look if someone has the time

**Comment by Heap412** [comment]
> If the user Deletes there current signature from Outlook Online and ensures both For New Messages, For Replies/Forwards = (No signature) this then oddly falls back to whatever is defined in the SignatureHTML property

This works but you have to delete all signatures then set For New Messages, For Replies/Forwards = (No signature), like he said and Save. This creates a new signature called "Signature Outlook Web" with the content of the SignatureHTML property. 

You can update/edit this signature in OWA but that doesn't update/edit the SignatureHTML property.
You can update/edit this signature with Powershell (cmdlet) but this takes a few minutes and doesn't work all the times ( I don't know why)

**Comment by andreaci** [comment]
> > If the user Deletes there current signature from Outlook Online and ensures both For New Messages, For Replies/Forwards = (No signature) this then oddly falls back to whatever is defined in the SignatureHTML property
> 
> This works but you have to delete all signatures then set For New Messages, For Replies/Forwards = (No signature), like he said and Save. This creates a new signature called "Signature Outlook Web" with the content of the SignatureHTML property.
> 
> You can update/edit this signature in OWA but that doesn't update/edit the SignatureHTML property. You can update/edit this signature with Powershell (cmdlet) but this takes a few minutes and doesn't work all the times ( I don't know why)

tried it and it doesn't work. 
actually it worked on one account and not on all the others.

**Comment by andreaci** [comment]
**OFFICIAL STATEMENT FROM MICROSOFT:**

the problem was introduced in July, with this feature (Outlook for windows roaming signatures):
https://www.microsoft.com/en-ie/microsoft-365/roadmap?filters=&searchterms=60371

the problem has been escalated and any news will be published here:

https://techcommunity.microsoft.com/t5/exchange/roaming-signatures-for-outlook-for-windows/m-p/1398915

unfortunately, no ETA

**Comment by beail** [comment]
> > > If the user Deletes there current signature from Outlook Online and ensures both For New Messages, For Replies/Forwards = (No signature) this then oddly falls back to whatever is defined in the SignatureHTML property
> > 
> > 
> > This works but you have to delete all signatures then set For New Messages, For Replies/Forwards = (No signature), like he said and Save. This creates a new signature called "Signature Outlook Web" with the content of the SignatureHTML property.
> > You can update/edit this signature in OWA but that doesn't update/edit the SignatureHTML property. You can update/edit this signature with Powershell (cmdlet) but this takes a few minutes and doesn't work all the times ( I don't know why)
> 
> tried it and it doesn't work. actually it worked on one account and not on all the others.

I found that this also seems to be impacted by the browser cache, maybe worth a clearing it

**Comment by leewisestamp** [comment]
> **OFFICIAL STATEMENT FROM MICROSOFT:**
> 
> the problem was introduced in July, with this feature (Outlook for windows roaming signatures): https://www.microsoft.com/en-ie/microsoft-365/roadmap?filters=&searchterms=60371
> 
> the problem has been escalated and any news will be published here:
> 
> https://techcommunity.microsoft.com/t5/exchange/roaming-signatures-for-outlook-for-windows/m-p/1398915
> 
> unfortunately, no ETA

No, I don't think this is related since this feature is explicitly for **_Outlook for Windows_** and not (explicitly) related to OWA.

**Comment by andreaci** [comment]
> > **OFFICIAL STATEMENT FROM MICROSOFT:**
> > the problem was introduced in July, with this feature (Outlook for windows roaming signatures): https://www.microsoft.com/en-ie/microsoft-365/roadmap?filters=&searchterms=60371
> > the problem has been escalated and any news will be published here:
> > https://techcommunity.microsoft.com/t5/exchange/roaming-signatures-for-outlook-for-windows/m-p/1398915
> > unfortunately, no ETA
> 
> No, I don't think this is related since the feature you referenced is explicitly for **_Outlook for Windows_** and not (explicitly) related to OWA.

outlook for win SHOULD sync signatures with OWA since july.

**Comment by leewisestamp** [comment]
> > > **OFFICIAL STATEMENT FROM MICROSOFT:**
> > > the problem was introduced in July, with this feature (Outlook for windows roaming signatures): https://www.microsoft.com/en-ie/microsoft-365/roadmap?filters=&searchterms=60371
> > > the problem has been escalated and any news will be published here:
> > > https://techcommunity.microsoft.com/t5/exchange/roaming-signatures-for-outlook-for-windows/m-p/1398915
> > > unfortunately, no ETA
> > 
> > 
> > No, I don't think this is related since the feature you referenced is explicitly for **_Outlook for Windows_** and not (explicitly) related to OWA.
> 
> outlook for win SHOULD sync signatures with OWA since july.

There is no mention in the roadmap item (or the techcommunity post) of OWA. Anyway, I don't think this is related to the issue we're all facing.

**Comment by yogkumgit** [comment]
@leewisestamp as advised by Chris multiple times, reporting support is the only way. We are here to improve  the documentation only. Proceed to close it. Thanks for your contribution.

**Comment by leewisestamp** [comment]
@yogkumgit And as I previously replied, I disagree that this is unrelated documentation. Until someone from Microsoft confirms one way or the other (i.e. if the product changed and therefore docs need updating, or if the product broke and therefore docs are good), this ticket should *remain open*!

https://github.com/MicrosoftDocs/office-docs-powershell/issues/8537#issuecomment-960574889

**Comment by boisrobe** [comment]
Already have the problem in july and it disapeared like it appears.
No problem since july...

My july ticket give me a poor explication and workaround (in french) : 
Voici le plan d’action/Plan d’analyse/Troubleshooting de MS : 
		○ 1. Naviguez dans Outlook en ligne -> icône de l'engrenage -> Afficher tous les paramètres d'Outlook.
		○ 2. Naviguer dans Courrier -> Composer et répondre
		○ 3. Définissez "Annuler l'envoi" à 0 et enregistrez.
(S'il est déjà à 0, donnez-lui une autre valeur et enregistrez, puis remettez-le à 0).

We have a formate signature design by our administration (gouvernment ministry).

The goal off a feature for outlook store in the cloud is cool, so we could have one script to configure OWA, Outlook desktop and outlook mobile. 
But it must be document.
And why is the multiple signature come before cloud storage off signature.
:-(

**Comment by TOH2F** [comment]
Hello,
is there a solution? I have the same problem

**Comment by alesysadmin75** [comment]
Hello,
same here.
It works individually "If the user Deletes there current signature from Outlook Online and ensures both For New Messages, For Replies/Forwards = (No signature) this then oddly falls back to whatever is defined in the SignatureHTML property."
But isn't a solution if you have hundreds of accounts to manage or updates.

I can't believe there is no documentation or solution about that :-(

**Comment by chrisda** [comment]
@alesysadmin75, @TOH2F and anyone else: if this change in functionality affects you, I highly recommend that you open a support case about it if you haven't already.

**Comment by wojcib** [comment]
Here is my advice for everyone that would like the cmdlet to work again. What I did:
In portal.office.com -> Administration -> Help -> Create new ticket.
In my particular case it took Microsoft 3 weeks, to disable 'multiple signatures' functionality. The powershell cmdlet works now.

There is no workaround. Only disabling mutliple signatures functionality by MS support.

**Comment by grryf** [comment]
So documentation is wrong for many people and we can't mention it in
documentation?

**Comment by grryf** [comment]
I have created ticket with MS support on 11th of January, but they still playing with me.

@wojcib what would you advise to ask Microsoft exaclty to do?

Description: issue described here: https://github.com/MicrosoftDocs/office-docs-powershell/issues/8537 [github.com] As a solution please disable multi signatures for our tenant.

**Comment by alesysadmin75** [comment]
I opened the ticket just 1 week ago.
Microsoft just called me. They disabled 'Roaming signature' on our tenant and they confirmed it as only workaround by now.

**Comment by caspChristian** [comment]
@yogkumgit why is this issue closed when there has been no update to the documentation?
Also no explanation on why it is closed without being addressed. 🤬

@chrisda Feel free to reopen again. ;)

**Comment by leewisestamp** [comment]
> @yogkumgit why is this issue closed when there has been no update to the documentation? Also no explanation on why it is closed without being addressed. 🤬
> 
> @chrisda Feel free to reopen again. ;)

Agree 100%

**Comment by tdherring** [comment]
Are there any updates on this or progress towards a Graph API alternative? We have been needing to use this cmdlet for months and there's seemingly no movement from Microsoft's end! When I contacted support I was told there would be a solution soon, so it wouldn't be necessary to disable roaming signatures (this was a couple of months ago).

**Comment by grryf** [comment]
Support just disabled roaming signatures for our tenant some time ago. It
works since.

**Comment by andreaci** [comment]
Support can disable the roaming signatures function.
They told me last week that it will be completed (and hopefully working) in July 2022.
They have no info on PowerShell api

**Comment by caspChristian** [comment]
The main issue here is still that there is no documentation about this cmdlet not working (in some cases?), and explanation as to why. MS claims this is not in a public release, and that disabling roaming signatures "resolves" the issue, totally ignoring the fact that any existing signatures seems to be lost in the process.

Real fix should be out when roaming signatures is GA, which I think it already is, so confusing! https://www.microsoft.com/en-in/microsoft-365/roadmap?filters=&searchterms=60371

**Comment by chrisda** [comment]
I added the following note to the top of the SignatureHtml parameter description:

> **Note**: This parameter doesn't work if the Outlook roaming signatures feature is enabled in your organization. Currently, the only way to make this parameter work again is to open a support ticket and ask to have Outlook roaming signatures disabled in your organization.

If/when there's ever a way to disable Outlook roaming signatures yourself (or any other fix), I'll updated the note.

OK to close this issue now?

**Comment by caspChristian** [comment]
Thank you @chrisda for the update, good to finally have this documented, but what took so long? ;)
Still tho, why is roaming profiles even deployed when there is no cmdlets to manage them? When contacting support they just refer to the roadmap that roaming profiles will be deployed later in 2022 and that it at that time will be cmdlets available to manager signatures. but as we can see here, roaming profiles are already deployed. I think that roaming profiles should not be enabled as it has been. Disable roaming profiles for everyone, or have working cmdlets, do not enable roaming profiles again until cmdlets is available.

**Comment by leewisestamp** [comment]
> I added the following note to the top of the SignatureHtml parameter description:
> 
> > **Note**: This parameter doesn't work if the Outlook roaming signatures feature is enabled in your organization. Currently, the only way to make this parameter work again is to open a support ticket and ask to have Outlook roaming signatures disabled in your organization.
> 
> If/when there's ever a way to disable Outlook roaming signatures yourself (or any other fix), I'll updated the note.
> 
> OK to close this issue now?

Thanks, @chrisda. I think your comment suffices for this issue. Frustrating that it took over six months to get resolved and that we were left without a real fix to the original functionality. But, at least the documentation now accurately reflects what to expect. Cheers.

**Comment by hkusulja** [comment]
Ok, any progress on ability to set signatures for mailbox using powershell or graph api when using roaming signatures? It is July 2022. If this is a new feature, what is number # in Microsoft 365 roadmap?

> Outlook for Windows: Signature cloud settings
> Your Signature settings are stored in the cloud, so your experience is consistent when you access Outlook for Windows on any computer. This feature has been delayed and will only be available in Beta Channel.  
> Feature ID: 60371
> Preview: September 2022
> GA: October 2022
https://www.microsoft.com/en-us/microsoft-365/roadmap?searchterms=60371

**Comment by beail** [comment]
Hello again, just an update

We have had the roaming feature disabled and can confirm this cmdlet now works, also confirmed Outlook roaming signatures feature is now disabled on the tenant and advised it will impact the release of the feature.

Findings however are contrary, when using the 'New' Outlook the cmdlet set signatures are present on the client (implying a roaming capability?) can the documentation at either clarify what the Outlook roaming signatures feature is and what disbaling it impacts or that stipulate that this cmdlet works when using the 'New' Outlook client?

**Comment by hkusulja** [comment]
@beail, this feature is now GA, so it is not anymore preview. It can be manually enabled on the tenant level still. Once enabled, Outlook Web App and Outlook desktop client uses the same signatures from cloud. I confirm that have several customers/tenants with this feature enabled and fully working.

What needs to be updated is the powershell and documentation for SignautreHtml parameter.

**Comment by chrisda** [comment]
OK. Per #10111 I removed the note.

**Comment by YanFonseca1** [comment]
Hi, 

Just to leave a quick feedback about this issue.

I was trying to use this command last week, October 28 to November 2, and it wasn't working.

Opened a ticket with support and in 1 week they disable Roaming for me, and works now.

**Comment by hkusulja** [comment]
I am not sure is it resolved or not.
Outlook Roaming is not anymore in beta, that is for sure.
But what is not sure, how to use PowerShell to set signature, when using Roaming on tenant features.
I believe this needs inspection from product group and update in docs still. For example in docs is not clear how to create multiple signatures for particular user and managed them. (again, this is not in beta anymore and is GA)

**Comment by jkrrv** [comment]
I'm confused. I had support disable roaming signatures in my organization a few months ago. Can I now have them re-enabled, but now be able to set SignatureHtml content via PowerShell? This is a pretty messy rollout.

**Comment by tdannecy** [comment]
Any updates on this issue? I am still unable to set an email signature using the `Set-MailboxMessageConfiguration` command.

> I added the following note to the top of the SignatureHtml parameter description:
> 
> > Note: This parameter doesn't work if the Outlook roaming signatures feature is enabled in your organization. Currently, the only way to make this parameter work again is to open a support ticket and ask to have Outlook roaming signatures disabled in your organization.
> 
> If/when there's ever a way to disable Outlook roaming signatures yourself (or any other fix), I'll updated the note.
> 
> OK to close this issue now?

The documentation page for `Set-MailboxMessageConfig` does not have the warning message: https://learn.microsoft.com/en-us/powershell/module/exchange/set-mailboxmessageconfiguration?view=exchange-ps

**Comment by chrisda** [comment]
@tdannecy, I was told to remove that note per this issue #10111. Was that guy wrong?

**Comment by jkrrv** [comment]
It seems that the only way to set signatures via PowerShell is still to disable roaming profiles, and it would seem that the only way to do that is to contact support and wait for them to figure out what you're talking about. (Especially now that it's not in the documentation.)

But ultimately the hope is that the PowerShell command or its underlying API are fixed such that they start to work with roaming profiles.

There are several problems:
1. Can't set signatures when roaming profiles are enabled.
2. The tenant admin can't readily see whether roaming profiles are enabled, and certainly can't control that setting.
3. If or when these issues are fixed, I'm not sure how we tenant admins would know about the fix, nor how to test it without a weeks-long dance with support.

________________________________

**Comment by caspChristian** [comment]
If this for whatever reason this doesn't currently work (as Microsoft promised it would over a year ago) then Microsoft really really needs to get their act together and get this fixed. [Case #:30105210]

Not to mention that removing these details if it doesn't work, If Microsoft can't confirm this working @chrisda ?? then we need a revert of the documentation quickly, and then we need to have this fixed (again this was guaranteed to be working by Microsoft when it was GA, even tho MS already have made this feature GA and then taken that back)

Support ticket is the only way to switch back from roaming signatures.

**Comment by chrisda** [comment]
OK. I'll put the note back in.

> **Note**: This parameter doesn't work if the Outlook roaming signatures feature is enabled in your organization. Currently, the only way to make this parameter work again is to open a support ticket and ask to have Outlook roaming signatures disabled in your organization.

**Comment by chrisda** [comment]
Does the registry setting discussed in [this topic](https://support.microsoft.com/office/outlook-roaming-signatures-420c2995-1f57-4291-9004-8f6f97c54d15) offer any relief from having to open a support ticket?

**Comment by GruberMarkus** [comment]
> Does the registry setting discussed in [this topic](https://support.microsoft.com/office/outlook-roaming-signatures-420c2995-1f57-4291-9004-8f6f97c54d15) offer any relief from having to open a support ticket?

No, setting DisableRoamingSignaturesTemporaryToggle in the registry only changes the behavior of Outlook on Windows, but has no influence on Outlook Web and if roaming signatures are enabled in Exchange Online at tenant or mailbox level.

**Comment by hkusulja** [comment]
@chrisda please reopen this GitHub issue since it is not resolved. What did product group say about this PowerShell cmdlet now when the roaming signatures are GA ?

**Comment by chrisda** [comment]
OK Original statement added back. @hkusulja, I have nothing to report.

**Comment by hkusulja** [comment]
@chrisda previously it was stated that this limitation is temporary until outlook roaming signatures are not GA, now they are GA for months but still limitation via PowerShell persists. I believe that then Exchange Online - API and PowerShell module needs to be updated (as product group feedback). Just to be on safe side, sharing now also this here - https://feedbackportal.microsoft.com/feedback/idea/46c270e4-7981-ed11-a81b-000d3ae32cd0

**Comment by chrisda** [comment]
@hkusulja, all of what you're saying sounds good, but all I can do here is fix documentation for the Set-MailboxMessageConfiguration cmdlet reference topic, which should now (again) be accurate.

If you haven't already, I highly encourage you and anyone else affected to open a support ticket about this. The product teams see this information.

**Comment by jkrrv** [comment]
Case #34380403‎

**Comment by hkusulja** [comment]
@chrisda Microsoft support ticket does not help. They just point to this documentation and statement that is there, and do not go further, nor the product group, or without any progress or answer.

**Comment by chrisda** [comment]
I realize there probably won't be immediate gratification, but the volume of support tickets about an issue is noticed by the people who need to see it.

**Comment by Kafeton** [comment]
Our signature scripts are broken as well. 

- The doc still states to contact support for the workaround 
https://learn.microsoft.com/en-us/powershell/module/exchange/set-mailboxmessageconfiguration?view=exchange-ps#examples
- The post on feedbackportal only contains outdated information about the roadmap
https://feedbackportal.microsoft.com/feedback/idea/46c270e4-7981-ed11-a81b-000d3ae32cd0
- On the roadmap the entry is in preview again with GA: April 2023
https://www.microsoft.com/en-us/microsoft-365/roadmap?searchterms=60371&filters=&searchterms=60371

And can we re-open this github bug so people coming here have the correct information without reading the whole thread? 
I opened a support ticket with all the above information and was immediatly told this is a known issue and that they will notify me as soon as it is resolved.

**Comment by chrisfsfs** [comment]
Wanted to add in here and say this has been broken for our thousands of users for a few months now, at least.

We would push out the HTML signatures via PowerShell script, which would update the OWA signatures for our users. However, we estimated ever since this has switched to the multiple signature option/dropdown, the signatures do not show or apply correctly.

If we open another user's mailbox via the OWA option, the other user's signature shows correctly (no multiple signature dropdown is shown). However, if that user, or even myself, try to view our own signature, it does not show the recently applied signature, or any at all in most cases (and it shows the single signature option). This leads me to believe the multiple signature feature clearly is broken.

Support ticket has been open for well over a month with constant back and forth, logs, screen captures, calls, etc. but with no resolution or any ideas. I even requested that roaming signatures are turned off for our entire organization on their end, and the confirmed with the engineers that they turned it off. No resolution yet...

**Comment by gabrielpereira1999** [comment]
I opened a ticket asking the feature to be disabled on our org on march 8th, even linked the documentation note and this thread. Had to give them lots of info and still no one even tried to disable the feature.

**Comment by badgeror** [comment]
Just to add to the fun, we're having the same issues.
On one of our tenants Microsoft disabled roaming signatures and we're able to sucessfuly deploy the signatures, however we're still having issues on another tenant, although it briefly worked after MS disabled the roaming, it appears to have stopped for some reason. MS Support on this most recent problem seems not able to resolve the issue, instead wanting a remote session which won't help...

**Comment by jnickell** [comment]
Wanted to add for the thread that today Support is telling me that disabling Outlook Roaming Signatures for OWA is no longer possible. It's "enabled globally" and for some sort of security reasons it cannot be disabled. No guidance was provided on how we are to administer signatures at this time.

**Comment by caspChristian** [comment]
@chrisda It's kind of weird that we cant have this working, it is way past the date of promise from support that this should have been fixed, especially if we now even cant disable roaming profiles!

**Comment by badgeror** [comment]
> Wanted to add for the thread that today Support is telling me that disabling Outlook Roaming Signatures for OWA is no longer possible. It's "enabled globally" and for some sort of security reasons it cannot be disabled. No guidance was provided on how we are to administer signatures at this time.

This is frustrating :/ We've still got a call open with MS about this so wonder if we'll eventually get the same response

**Comment by hkusulja** [comment]
@chrisda why is this not reopened , again, workaround / reverting back does not help anymore. Please ask product group about this documentation.

**Comment by WardenHub** [comment]
With the new outlook version fixing this is imperative, I cant ask the entire company to copy paste their signatures from before!

**Comment by chrisfsfs** [comment]
Thank you for reopening this. Hopefully this actually will make some progress!

**Comment by chrisda** [comment]
Keep in mind, this isn't a support ticket. You, me, or anyone else can close and open GitHub issues. The product team does not watch these issues; only guys like me who are interested in documentation updates do.

**Comment by chrisfsfs** [comment]
> Keep in mind, this isn't a support ticket. You, me, or anyone else can close and open GitHub issues. The product team does not watch these issues; only guys like me who are interested in documentation updates do.

Got it. We have had a support ticket opened for months with no progress. Good to know about this thread and keeping an eye on it for any progress that might be made.

**Comment by dunck** [comment]
Many are stating that they have had tickets open with no progress, or have had trouble communicating what needs to be done. I was able to have this change made for my org with the following request:

> We can no longer use the `Set-MailboxMessageConfiguration` cmdlet to apply HTML signatures to mailboxes. This issue is well known and further explained here. https://github.com/MicrosoftDocs/office-docs-powershell/issues/8537. I am kindly requesting "Roaming Signatures" be disabled for our organisation as per advice.

Though I fully understand that there exists different levels of understanding between support units.

**Comment by dunck** [comment]
Many are stating that they have had tickets open with no progress, or have had trouble communicating what needs to be done. I was able to have this change made for my org with the following request:

> We can no longer use the `Set-MailboxMessageConfiguration` cmdlet to apply HTML signatures to mailboxes. This issue is well known and further explained here. https://github.com/MicrosoftDocs/office-docs-powershell/issues/8537. I am kindly requesting "Roaming Signatures" be disabled for our organisation as per advice.

Though I fully understand that perhaps there is a different level of understanding between support units.

**Comment by Bestora** [comment]
> Many are stating that they have had tickets open with no progress, or have had trouble communicating what needs to be done. [...]
> Though I fully understand that there exists different levels of understanding between support units.

Same here, we dropped our support request, because the support won't read this issue thread. They were only helping with looking at our powershell script and that's it... over 5 calls... No result at the end.

**Comment by crazycatdevs** [comment]
With the future version of Outlook, `Set-MailboxMessageConfiguration` is mandatory: we can't tell our users to change by the hand their own signature.
And knowing my users, our support team will have to do that for them, I can't ask that to my team.

**Comment by chrisfsfs** [comment]
Just wanted to chime back in here. I had two tickets opened (one for each of our tenants), and support FINALLY was able to opt us out of roaming signatures and now PowerShell command works perfectly! The roaming signature option is removed from our option for all users and the command pushed out successfully.

I really hope the documentation for this gets better, and for those of you looking to get this fixed, be persistent! It amazing how the new tenant we spun up got resolved within a week, but the other older tenant it is going on 2+ months and they're going in circles with "checking the logs" - yikes!

Before support opted us out:
![image](https://github.com/MicrosoftDocs/office-docs-powershell/assets/127867788/af151571-ed65-4a2c-92ac-937497fdf8c1)

After support opted us out:
![image](https://github.com/MicrosoftDocs/office-docs-powershell/assets/127867788/557840fd-b2f1-452a-93b9-8c97195226f3)

![image](https://github.com/MicrosoftDocs/office-docs-powershell/assets/127867788/66a1dcc7-90fd-4ea0-988b-a8d746c54604)

Case 36208070

**Comment by jgrigera** [comment]
Can we point out that this is no real resolution to the issue? It is obvious that there should be an API update to be able to change roaming signatures. If this was an open source project it would have been done ages ago.

**Comment by caspChristian** [comment]
@yogkumgit Can you please explain in what way this issue has been resolved? If not why was it closed?

**Comment by chrisda** [comment]
I reopend the issue, for what that's worth.

Fundamentally, this forum is for fixing **documentation** issues in Exchange-related PowerShell cmdlet reference topics (which includes Security & Compliance PowerShell). I realize the nature and duration of the problem has morphed this GitHub issue into something else, which I'm not insensitive to.

The documentation end of this problem has been fixed (then modified, then fixed again) a long time ago by stating the current reality in the [SignatureHtml parameter description](https://learn.microsoft.com/powershell/module/exchange/set-mailboxmessageconfiguration#-signaturehtml):

> **Note**: This parameter doesn't work if the Outlook roaming signatures feature is enabled in your organization. Currently, the only way to make this parameter work again is to open a support ticket and ask to have Outlook roaming signatures disabled in your organization.

So, from the documentation end, this issue has been 'fixed', which is why this issue keeps getting closed.

My last big sweep in April of comparing what cmdlets/parameters are available to me in a plain Microsoft 365 E5 tenant vs. what's documented revealed no new cmdlets that looked like they would help with this issue. I'll probably due another sweep in July.

But, the fix for this issue is likely to come from **outside** of Exchange-related PowerShell, because the Outlook roaming signatures feature itself appears to operate outside of Exchange-related PowerShell (which explains why turning the feature off makes the world right again).

**Comment by crazycatdevs** [comment]
My little experience with that: I had to spend 3 weeks with the MS support (first mail explaining why I need to have this option activated on my domains, with links to issue, KB and other docs).
The support explained me how to add signature in my own box, .... Finally, when I answer that I knew that disabling this option will also disable the hability to have several signatures (I don't care about it, I don't want it, it's company preformated signature and I'll kill the first one which think to modify it :D), they uncheck the f*cking checkbox and my system works.

imho, the issue is that domain admin can't disable this feature, using powershell or web interface.

**Comment by yogkumgit** [comment]
@leewisestamp Hope the comment by Chris is helpful for you. If you see a documentation update is required, please feel free to open an issue for the same. We proceed here to close it. Thanks for taking out some time to open the issue. Appreciate and encourage you to do the same in future also.

**Comment by WardenHub** [comment]
_Today on navigating Microsoft's maze of annoyances_

**Comment by GruberMarkus** [comment]
Set-MailboxMessageConfiguration still does not work and not all tenants or users see the new feature in Outlook Web.

**I opened a support call last week, because my different Outlook clients already sync signatures but they do not show up in Outlook Web. Result: "This is expected behavior at the moment. They estimate the rollout to Outlook Web will finish by the end of July. We can't opt-in your tenant to bring the feature sooner."**


As maintainer of the free and open-source core of Set-OutlookSignatures (https://github.com/GruberMarkus/Set-OutlookSignatures), I want to additionally share something that is good news for at least some of us in this thread: The first signature management solution for Outlook with support for roaming signatures has been released. The feature is only available in the commerical Benefactor Circle add-on and is marked experimental as there is no API officially released by Microsoft, but it works stable in the trial version: https://explicitconsulting.at/open-source/set-outlooksignatures/
The interesting thing is that the solution is independent of the Outlook version used. It downloads roaming signatures, treats them just like local ones during the signature update process, and then uploads all locally available signatures. The old Outlook Web single signature feature and the new multiple roaming signatures feature are supported at the same time, and it even does not matter if Outlook is configured to use roaming signatures or not.

**Comment by hkusulja** [comment]
@GruberMarkus but how can this feature work if there is no API ?

**Comment by GruberMarkus** [comment]
> @GruberMarkus but how can this feature work if there is no API ?

By watching what the client really does in detail. I am sorry, I can't go into more details.

**Comment by hkusulja** [comment]
therfore, asking microsoft / product group to document this here

**Comment by adrwh** [comment]
I intercepted the HTTP PATCH and inserted the HTML signature as needed, which works as expected.  Obviously though this is individual solution, not great for mass management.

**Comment by chrisda** [comment]
Greetings. A new parameter named [PostponeRoamingSignaturesUntilLater](https://learn.microsoft.com/powershell/module/exchange/set-organizationconfig#-postponeroamingsignaturesuntillater) has been added to the Set-OrganizationConfig cmdlet that allows admins to turn off Outlook roaming signatures themselves without the need to open a support ticket. Setting this parameter to $true allows the signature-related parameters on the Set-MailboxMessageConfiguration cmdlet to work as expected.

It's being called a 'temporary' parameter because there will eventually be API access for admins to manage/configure Outlook roaming signatures directly, so there will be no need to turn off Outlook roaming signatures and/or use the signature-related parameters in Set-MailboxMessageConfiguration. After API access to roaming signatures is available, the PostponeRoamingSignaturesUntilLater parameter on Set-OrganizationConfig will be deprecated.

I've [updated the signature-related parameter descriptions](https://github.com/MicrosoftDocs/office-docs-powershell/pull/11213/files) in Set-MailboxMessageConfiguration to mention this new parameter.

I'm happy to leave this issue open until the new API is available.

**Comment by Sky05727** [comment]
Hi chrisda

Thank you for the update! This sounds promising.

I've just set the PostponeRoamingSignaturesUntilLater function in our organization, but the signatures that are set with "Set-MailboxMessageConfiguration" are still not added to Outlook.com.

Is it because the function is not fully working yet?
I saw this comment from Microsoft:
Note: This parameter is in the process of being rolled out. The rollout is expected to be completed by mid-November 2023.

**Comment by hkusulja** [comment]
@yogkumgit  why is this added as complete? is documentation updated on how can you update signature settings using graph api or exchange powershell module for roaming signatures?

**Comment by WardenHub** [comment]
Dont think they did @hkusulja  [this is the only documentation I could find](https://learn.microsoft.com/en-us/powershell/module/exchange/set-mailboxmessageconfiguration?view=exchange-ps#-autoaddsignature)

Graph api acces and set- command info should definetly be added imo.

**Comment by hkusulja** [comment]
question is how to set signatures when using roaming signatures, not without them or by disabling them :)

**Comment by chrisda** [comment]
Reopening.

**Comment by officedocspr5** [comment]
To make it easier for you to submit feedback on articles on learn.microsoft.com, we're transitioning our feedback system from GitHub Issues to a new experience. 

As part of the transition, this GitHub Issue will be moved to a private repository. We're moving Issues to another repository so we can continue working on Issues that were open at the time of the transition. When this Issue is moved, you'll no longer be able to access it.

If you want to provide additional information before this Issue is moved, please update this Issue before December 15th, 2023.

With the new experience, you no longer need to sign in to GitHub to enter and submit your feedback. Instead, you can choose directly on each article's page whether the article was helpful. Then you can then choose one or more reasons for your feedback and optionally provide additional context before you select **Submit**.

Here's what the new experience looks like.

**Note**: The new experience is being rolled out across learn.microsoft.com in phases. If you don't see the new experience on an article, please check back later.

First, select whether the article was helpful:

![Image showing a dialog asking if the article was helpful with yes and no answers.](https://learn.microsoft.com/Office/media/internal/new-feedback-yes-no.jpg)

Then, choose at least one reason for your feedback and optionally provide additional details about your feedback:

| Article was helpful | Article was unhelpful |
|:--|:---|
|![Image showing a dialog asking how the article was helpful with several options.](https://learn.microsoft.com/Office/media/internal/new-feedback-helpful.jpg) | ![Image showing a dialog asking how the article wasn't helpful with several options.](https://learn.microsoft.com/Office/media/internal/new-feedback-unhelpful.jpg)|

Finally, select **Submit** and you're done!

**Comment by bpienig** [comment]
This issue is still broken, and PS does not update Office 365 Signatures at all.  Any update on this?  Is there a spot where we can see the status on this bug / feature / issue?
